shared/lib_pv_io_manager.cpp

```diff
@@ -1488,7 +1488,7 @@ void Inverter_IO::setupSharedInverter(compute_module* cm, SharedInverter* a_shar
         thermalDerateCurves.push_back(row);
     }
     int err = sharedInverter->setTempDerateCurves(thermalDerateCurves);
-    if (err > 1) {
-        throw exec_error("pvsamv1", "Inverter temperature derate curve " + util::to_string((int)(-err - 1)) + " is invalid.");
+    if (err > 0) {
+        throw exec_error("pvsamv1", "Inverter temperature derate curve row " + util::to_string((int)(err - 1)) + " is invalid.");
```

Comment:
@dguittet This correction caused a (correct) test failure in PVYield where the inverter temperature derate curves are not defined for the Inverter Mermoud Lejeune Model. Should this model include the option for temperature derates. How should this be handled?

Reply:
I think you can throw an exec_error for when that inverter model is selected and temperature derate curves are provided. That inverter model is only accessible via the SDK and so the UI defaults won't be contradictory to throwing that error.

Reply:
I understand that these inputs aren't needed for that particular inverter model but don't we still want to allow the model to continue running even if the defaults for the derate curves are inputs? Adding an else loop to get out of the loop before the setTempDerateCurves works for the tests. Would this cause any other errors?

```
if (inverterType == 0) // cec database
    {
        inv_tdc = cm->as_matrix("inv_tdc_cec_db");
    }
    else if (inverterType == 1) // datasheet data
    {
        inv_tdc = cm->as_matrix("inv_tdc_ds");
    }
    else if (inverterType == 2) // partload curve
    {
        inv_tdc = cm->as_matrix("inv_tdc_plc");
    }
    else if (inverterType == 3) // coefficient generator
    {
        inv_tdc = cm->as_matrix("inv_tdc_cec_cg");
    }
    else
    {
        return;
    }
```

Reply:
The default curves could be inputs to this inverter model if the user first used the code generator feature or loaded the defaults in PySAM and then changed the `inverter_model`. But in that case, rather than just ignoring the temp derate curves, I think it's safer from the user impact perspective to throw an error. Then they can remove the curves. If we just don't run the temperature derate, that could be more confusing as to why their simulation results are inconsistent with their inputs.